### PR TITLE
setup a demo of standalone repl usage

### PR DESCRIPTION
**Pull request by hackape**
so that people can `npm start` to see it in action

### CODE REVIEW

**Comment by antony** [comment]
This is great - I'm not 100% sure about the notion of including it within the project itself, but I don't really have an alternate solution right now. I'll wait for further comments from other maintainers on this approach.

**Comment by Conduitry** [comment]
I was just taking a look at this, and I think so far my main two points of feedback are

- do we want to make this a separate subproject, so that `demo` has its own package.json, rollup.config.js, etc? It might be a little tidier to have these things separate
- is all that CSS necessary?

This is great though, and definitely something we've been in dire need of for a while.

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "main": "src/Repl.svelte",
   "svelte": "src/Repl.svelte",
   "scripts": {
+    "start": "rollup -w -c demo/rollup.config.js",
```
Comment (antony) [inline package.json:8]: Going to suggest that we prepend `npm run srcdoc` to this too, since it doesn't work without it.

**Comment by pngwn** [comment]
I think making this as standalone as possible, with its own `package.json` etc would make sense. That might make it easier for people to get started, although many 'examples' directories in other projects don't tend to do this.

@Conduitry I don't know if all of that CSS is strictly necessary but styling is an issue with the REPL, when I have used it in the past it breaks if you don't add a bunch of styling from the svelte site. This is probably different conversation about how to make the REPL them-able / style-able / more reusable. There are few improvements we could make outside of just the styling (complicated somewhat by the use of workers). 

This is a pretty complex component and has the potential to be very valuable as the Svelte community grows but usage is a bit limited at the moment.

**Comment by hackape** [comment]
Thank you guys for all the feedbacks.

@antony thanks for reviewing!

> Going to suggest that we prepend `npm run srcdoc` to this too, since it doesn't work without it.

You're right, I missed that part.

But I must ask about the background why this `scripts/build-srcdoc.js` approach is preferred in the first place? I wasn't able to run that script successfully because I live in China and am blocked from `raw.githubusercontent.com` domain. So I ended up resolved it by hand.

Seems like it just downloads a css and concat it to an html, both files don't seem to change drastically. Wound't it make more sense to include that css inside the repo to make it more self-contained? 

My suggested approach: include the css right into `src/Output/srcdoc/index.html`, then use [`rollup-plugin-string`](https://github.com/TrySound/rollup-plugin-string/blob/master/index.js) to import that html, and ditch `build-srcdoc.js`.

**Comment by hackape** [comment]
@Conduitry Hi I'm quite new to svelte and don't really know the team/community members well. Looks like you're one of the core maintainers! Just wanna say I really appreciate your work to bring into this world an amazing new idea. Great job!

Response to your feedbacks:

> * do we want to make this a separate subproject, so that `demo` has its own package.json, rollup.config.js, etc? It might be a little tidier to have these things separate

I'm not 100% sure what you mean by `separate subproject`. If you mean a sub git module that'd be too much... Or a separate `package.json`? That'd require ppl to `npm install` twice. I personally feel a "sub folder" like it is currently is optimal. Don't see much value added to separate it any further.

> * is all that CSS necessary?

Believe me it looks terrible without those CSS 😂. So yes it's necessary.

**Comment by hackape** [comment]
@pngwn 

> This is a pretty complex component and has the potential to be very valuable as the Svelte community grows but usage is a bit limited at the moment.

I feel the same. I was thinking about converting this Repl component into a vscode extension feature (see https://github.com/sveltejs/language-tools/pull/183). But the current state of source code seems hard to make such conversion.

The interface is not very configurable. Like for example in vscode I would ditch the code editor part and just keep the `Output` component. But most parts are now tightly coupled cuz `getContext('REPL')` approach.

Plus as a typescript-spoiled kid I also feel uncomfortable to code without interface hinting 😂. So, if I were able to find time to eventually impl it, I would probably start afresh and harvest parts from this repo instead.

**Comment by thebells1111** [comment]
I'm wondering if catering to two different audiences may help. 

The solution by @hackape is great for those who want a working example they can view while they are editing the source files to contribute to `svelte/svelte-repl`. Keeping a `demo` folder would serve this purpose well. 

However, there are also those who just want a working example they can use in their own project, and having a stand alone REPL template would suit that purpose. Something as simple as typing
 `npx degit sveltejs/template-repl my-repl` would get them started.

**Comment by dimfeld** [comment]
Regarding the CSS, I got it to work and look nice on my own site with just setting these CSS variables on the REPL container element.

```css
.svelte-repl {
    background-color: #fff;
    --font: 'Inter', 'Open Sans', 'Helvetica', 'Verdana', sans-serif;
    --font-mono: 'Inconsolata', 'Menlo', 'Monaco', 'Consolas', 'Liberation Mono', 'Courier New', monospace;
    --prime: rgb(3, 102, 114);
    --second: #676778;
    --back-light: #f6fafd;
}
```

These values are my own of course. The official demo would probably want to use the values from the normal Svelte REPL.

**Comment by quantuminformation** [comment]
Any updates guys, I would like to use this but there are merge conflicts now. Do I just go with the src on this branch?

**Comment by FractalHQ** [comment]
I managed to modify the one from the mdsvex docs website and it works great.  I’d recommend checking it out.  Plus it has mdsvex voodoo built in!

**Comment by quantuminformation** [comment]
got a link?

**Comment by quantuminformation** [comment]
@dimfeld did you use the code given by @hackape ?

**Comment by dimfeld** [comment]
No, I can't remember the process I took to figure it out. But you might check out these files to see what I'm doing now:

My REPL wrapper component: https://github.com/dimfeld/website/blob/master/src/interactive/ReplWrapper.svelte

Rollup config to build the workers: https://github.com/dimfeld/website/blob/master/rollup-repl-workers.js

**Comment by quantuminformation** [comment]
@dimfeld  Is ReplWrapper.svelte pretty much self contained apart from the npm imports?

**Comment by dimfeld** [comment]
You'll have to rip out the part that tries to download the REPL source from my serverless function, and also replace the Tailwind styles with other CSS if you're not using that. Otherwise, probably self-contained? I've never tried to run it anywhere else so no guarantees :)

You can find examples all over my website. One such is at the bottom of this page: https://imfeld.dev/writing/svelte_domless_components

**Comment by quantuminformation** [comment]
I'm using this now, its fab

https://github.com/pngwn/REPLicant

**Comment by Rich-Harris** [comment]
Thank you for putting this together. We've moved `svelte-repl` to the https://github.com/sveltejs/sites/tree/master/packages/repl monorepo, where it's easy to run a demo app:

```
git clone git@github.com:sveltejs/sites.git
cd sites
pnpm install
cd packages/repl
pnpm dev
```

Closing this — will document these steps over on the monorepo

**Pull request closed without merging**